### PR TITLE
use taskcluster-worker-runner

### PR DESCRIPTION
**Pull request by aerickson**
see https://bugzilla.mozilla.org/show_bug.cgi?id=1602946

https://github.com/taskcluster/taskcluster-worker-runner

[x] add binary
[x] create config
[x] use binary to start g-w
[x] verify g-w config is identical
[x] cleanup testing/debugging code
[x] test
[x] upgrade to 1.0.4
[x] remove all debugging code and g-w config template
[x] retest

### CODE REVIEW

File: scripts/entrypoint.sh [hunk 1]
```diff
@@ -28,7 +28,8 @@ entrypoint.py
 cd $HOME
 generic-worker new-ed25519-keypair --file $ED25519_PRIVKEY
 chown worker $ED25519_PRIVKEY
-envsubst < $CONF_PATH/generic-worker.yml.template > $CONF_PATH/generic-worker.yml
+envsubst < $CONF_PATH/generic-worker.yml.template > $CONF_PATH/generic-worker-ORIG.yml
+envsubst < $CONF_PATH/worker-runner-config.yml.template > $CONF_PATH/worker-runner-config.yml
```
Comment (djmitche) [inline scripts/entrypoint.sh:31]: TIL envsubst!

**Review by djmitche — COMMENTED** [review]
This looks good!  I guess a second step would be to switch this to a `static` providerType?

**Comment by aerickson** [comment]
> This looks good! I guess a second step would be to switch this to a `static` providerType?

I'm open to that... then staticSecrets for each host would replace the taskcluster tokens we currently have for each pool? What worker-manager would these workers use?

Thanks,
Andy

**Comment by djmitche** [comment]
Precisely.

And, you would create new worker pools `proj-autophone/${TC_WORKER_TYPE}` in the firefox-ci worker-manager, and then call `createWorker` for each worker instance.

That will get you a few benefits immediately, notably the worker pools will show up in the UI with the rest of them, and you can see pending counts, etc.  And as we add a bit more functionality, your workers will also show up in the UI, and you'll be able to provide config to them via worker-manager, too.

File: Dockerfile [hunk 2]
```diff
@@ -80,6 +80,7 @@ ADD https://dl.google.com/android/repository/sdk-tools-linux-4333796.zip /builds
 ADD https://github.com/taskcluster/generic-worker/releases/download/v16.0.0/generic-worker-simple-linux-amd64 /usr/local/bin/generic-worker
 ADD https://github.com/taskcluster/livelog/releases/download/v1.1.0/livelog-linux-amd64 /usr/local/bin/livelog
 ADD https://github.com/taskcluster/taskcluster-proxy/releases/download/v5.1.0/taskcluster-proxy-linux-amd64 /usr/local/bin/taskcluster-proxy
+ADD https://github.com/taskcluster/taskcluster-worker-runner/releases/download/v1.0.2/start-worker-linux-amd64 /usr/local/bin/start-worker
```
Comment (bclary) [inline Dockerfile]: 12 days old with v1.0.4 available. Should we stay on the bleeding edge?

File: scripts/entrypoint.sh [hunk 3]
```diff
@@ -28,7 +28,8 @@ entrypoint.py
 cd $HOME
 generic-worker new-ed25519-keypair --file $ED25519_PRIVKEY
 chown worker $ED25519_PRIVKEY
-envsubst < $CONF_PATH/generic-worker.yml.template > $CONF_PATH/generic-worker.yml
+# envsubst < $CONF_PATH/generic-worker.yml.template > $CONF_PATH/generic-worker-ORIG.yml
```
Comment (bclary) [inline scripts/entrypoint.sh]: What's the benefit of keeping these in the file as comments?

File: scripts/run_gw.py [hunk 4]
```diff
@@ -31,10 +31,11 @@ def log_to_pt(message, print_to_screen=False):
         print(message)
 
 scriptvars_json_file = '/builds/taskcluster/scriptvars.json'
-gw_config_file = "/builds/taskcluster/generic-worker.yml"
+# gw_config_file = "/builds/taskcluster/generic-worker.yml"
+tc_worker_runner_config_file = "/builds/taskcluster/worker-runner-config.yml"
```
Comment (bclary) [inline scripts/run_gw.py:34]: ditto comments

File: scripts/run_gw.py [hunk 5]
```diff
@@ -31,10 +31,11 @@ def log_to_pt(message, print_to_screen=False):
...
 hostname = socket.gethostname()
 
-cmd_str = "generic-worker run --config %s" % gw_config_file
```
Comment (bclary) [inline scripts/run_gw.py:37]: But we don't leave a commented version of this line...

**Review by bclary — CHANGES_REQUESTED** [review]
Let's get the latest taskcluster-worker-runner and remove the commented code we are replacing unless we have a good reason to keep it.

File: Dockerfile (hunk 2 above)
Comment (djmitche) [reply Dockerfile]: Yes please - v1.0.4 adds a security check.

File: scripts/entrypoint.sh (hunk 3 above)
Comment (aerickson) [reply scripts/entrypoint.sh]: It makes it easier to ensure that the configs generated are identical in future testing.

File: Dockerfile (hunk 2 above)
Comment (aerickson) [reply Dockerfile]: Updated and retested.

**Review by bclary — APPROVED** [review]
r+ If the comments help you, by all means leave them in.

**Pull request merged**